modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/challenge/questions/ServerChallengeTestCase.java

```diff
@@ -0,0 +1,346 @@
+import com.google.gson.GsonBuilder;
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.testng.Assert;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+import org.wso2.identity.integration.test.rest.api.challenge.questions.model.ServerChallengeModel;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.restassured.RestAssured.given;
+
+/**
+ * This class is for ServerChallengeQuestion rest api test cases.
+ */
+public class ServerChallengeTestCase {
+
+
+    private final String username = "admin";
+    private final String password = "admin";
+    private static final String serverUrl = "https://localhost:9853";
+    private static final String resourcePath = "/api/server/v1/challenges";
+    private static final String questionId = "question7";
+    private final static String questionSetId = "challengeQuestion7";
+    private static final String locale = "en_US";
+    private static final String challengeQuestion = "What is your best friend name?";
+    private static final String questionSetIdAttribute = "questionSetId";
+    private static final String questionIdAttribute = "questionId";
+    private static final String questionAttribute = "question";
+    private static final String questionObjectAttribute = "questions";
+    private static final String localeAttribute = "locale";
+
+    /**
+     * RestAssured.baseURI, which sets the base URI statically.
+     */
+    @BeforeTest(alwaysRun = true)
+    public void testinit() {
+
+        RestAssured.baseURI = serverUrl;
+
+    }
+
+    /**
+     * The method is used to add new challenge question set.
+     *
+     * @throws JSONException thrown to indicate a problem with the JSON.
+     */
+    @Test
+    public void addsNewChallengeQuestionSetTest() throws JSONException {
+
+        ServerChallengeModel.Questions question = new ServerChallengeModel.Questions(locale,
+                challengeQuestion, questionId);
+        List<ServerChallengeModel.Questions> questions = new ArrayList<>();
+        questions.add(question);
+
+        List<ServerChallengeModel> serverChallenges = new ArrayList<>();
+        ServerChallengeModel serverChallengeRequest = new ServerChallengeModel(questionSetId, questions);
+        serverChallenges.add(serverChallengeRequest);
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        String jsonRequrst = gson.toJson(serverChallenges);
+
+        given()
+                .auth().preemptive().basic(username, password)
+                .contentType(ContentType.JSON)
+                .header("Accept", ContentType.JSON)
+                .body(jsonRequrst)
+                .log().all()
+                .when()
+                .post(resourcePath).
+                then().assertThat().statusCode(201).log().ifValidationFails();
```

Comment:
Can use constant for 201. Can use from org.apache.http.HttpStatus
Check other places for similar usage

Reply:
Modified with org.apache.http.HttpStatus constant